servicetalk-http-router-predicate/src/main/java/io/servicetalk/http/router/predicate/InOrderRouter.java

```diff
@@ -68,9 +68,9 @@
                 StreamingHttpService service = pair.service();
                 final HttpExecutionStrategy strategy = pair.routeStrategy();
                 if (strategy != null) {
-                    Executor se = strategy.executor();
-                    Executor executor = null == se ? ctx.executionContext().executor() : se;
-                    service = strategy.offloadService(executor, service);
+                    Executor executor = ctx.executionContext().executor();
+                    service = StreamingHttpService.offloadService(strategy,
+                            executor, Boolean.TRUE::booleanValue, service);
```

Comment:
`Boolean.TRUE::booleanValue` -> should this be sensitive to `IoThread`?

Reply:
I had left those cases alone after refactoring to use the same implementation for both `OffloadFilter` and the removed `HttpExecutionStrategy::offloadService`. I have converted them now and they work correctly so they are now conditionally offloading.